src/components/Notifications/NotificationEmailCard/NotificationEmailCard.tsx

```diff
@@ -0,0 +1,140 @@
+      setIsValidEmail(false)
+    }
+  }, [email])
+
+  const handleOnChangeEmail = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setEmail(event.target.value)
+    },
+    [setEmail]
+  )
+
+  const handleOnChangeNotificationSetting = useCallback(
+    (_: React.ChangeEvent<HTMLInputElement>, checked: boolean) => {
+      if (!isLoading) {
+        const subscriptionDetailsChanged = {
+          ...subscriptionDetails,
+          isIgnoringAllEmail: !checked
+        }
+        onChangeNotificationSetting(objectToSnake(subscriptionDetailsChanged))
+      }
+    },
+    [subscriptionDetails, onChangeNotificationSetting, isLoading]
+  )
+
+  const descriptionText = useMemo(() => {
+    if (!emailProp && !unconfirmedEmail && !hasConfirmEmail) {
+      return t('settings.notifications.email.description.without_email')
+    } else if (unconfirmedEmail) {
+      return t('settings.notifications.email.description.pending_approval')
+    } else if (hasConfirmEmail) {
+      return t('settings.notifications.email.description.confirmed')
+    }
+    return t('settings.notifications.email.description.with_email')
+  }, [emailProp, unconfirmedEmail, hasConfirmEmail])
+
+  const buttonText = useMemo(() => {
+    if (!emailProp && !unconfirmedEmail && !hasConfirmEmail) {
+      return t('settings.notifications.email.button_submit_label')
+    } else if (unconfirmedEmail) {
+      return t('settings.notifications.email.button_resend_label')
+    }
+    return t('settings.notifications.email.button_edit_label')
+  }, [emailProp, unconfirmedEmail, hasConfirmEmail])
+
+  return (
+    <>
+      <Card>
+        <TitleContainer>
+          <Title variant="h6" data-testid={NOTIFICATION_EMAIL_CARD_TITLE_TEST_ID}>
+            {t('settings.notifications.email.title')}
+
+            {(unconfirmedEmail || hasConfirmEmail) && (
+              <SpanUnconfirmedEmail confirmed={hasConfirmEmail} data-testid={NOTIFICATION_EMAIL_CARD_UNCONFIRMED_TEST_ID}>
+                {hasConfirmEmail ? t('settings.notifications.email.confirmed') : t('settings.notifications.email.pending_approval')}
+              </SpanUnconfirmedEmail>
+            )}
+          </Title>
+          {!!emailProp && (
+            <Switch
+              onChange={handleOnChangeNotificationSetting}
+              checked={!isIgnoringAllEmail}
+              data-testid={NOTIFICATION_EMAIL_CARD_SWITCH_TEST_ID}
+            />
+          )}
+        </TitleContainer>
+        <Description data-testid={NOTIFICATION_EMAIL_CARD_DESCRIPTION_TEST_ID}>{descriptionText}</Description>
+        <InputContainer>
+          <TextField
+            type="email"
+            placeholder={t('settings.notifications.email.placeholder')}
+            value={email}
+            onChange={handleOnChangeEmail}
+            variant="outlined"
+            error={!isValidEmail}
+            helperText={!isValidEmail && t('settings.notifications.email.invalid_email')}
+            disabled={isLoading}
```

Comment:
It seems that there's no distinction between `isSavingEmail` or `isLoading` in terms of usability. What if we simplify the component by only receiving `isLoading`?
The `isLoading` prop can contain the `isSavingEmail` prop in its logic.

Reply:
You are right. I change the loading behaviour and forgot that `isSavingEmail`